cpp/tests/test_abm_world.cpp

```diff
@@ -305,20 +306,45 @@ TEST(TestWorld, evolveMigration)
         auto world       = mio::abm::World(params);
         auto home_id     = world.add_location(mio::abm::LocationType::Home);
         auto work_id     = world.add_location(mio::abm::LocationType::Work);
+        auto hospital_id = world.add_location(mio::abm::LocationType::Hospital);
         auto icu_id      = world.add_location(mio::abm::LocationType::ICU);
 
         auto& p_dead = add_test_person(world, icu_id, mio::abm::AgeGroup::Age60to79, mio::abm::InfectionState::Dead, t);
+        auto& p_critical = add_test_person(world, hospital_id, mio::abm::AgeGroup::Age60to79,
+                                           mio::abm::InfectionState::InfectedCritical, t);
+        p_dead.set_assigned_location(icu_id);
         p_dead.set_assigned_location(home_id);
         p_dead.set_assigned_location(work_id);
+        p_critical.set_assigned_location(hospital_id);
+        p_critical.set_assigned_location(icu_id);
```

Comment:
I think that this person does not start in hospital as it is in Critical (ICU) state. Am I wrong? If not, please correct this.

Reply:
Do you mean that it SHOULD not start there? I think the starting location to that person is not really relevant, but we can surely change it.

Reply:
Should in the sense @khoanguyen-dev intended it. As also the name suggests.

Reply:
Yes, i do agree. I will change the name to reflect this and assigned the new infection later.